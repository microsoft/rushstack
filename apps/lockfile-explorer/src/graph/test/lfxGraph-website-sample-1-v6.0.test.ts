// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import type { IJsonLfxWorkspace } from '../../../build/lfx-shared';

import * as graphTestHelpers from './graphTestHelpers';

export const workspace: IJsonLfxWorkspace = {
  workspaceRootFullPath: '/repo',
  pnpmLockfilePath: 'common/temp/pnpm-lock.yaml',
  pnpmLockfileFolder: 'common/temp',
  pnpmfilePath: 'common/temp/.pnpmfile.cjs',
  rushConfig: {
    rushVersion: '5.158.1',
    subspaceName: '',
    rushPnpmfilePath: 'common/config/.pnpmcfile.cjs'
  }
};

describe('lfxGraph-website-sample-1-v6.0', () => {
  it('loads a workspace', async () => {
    const serializedYaml: string = await graphTestHelpers.loadAndSerializeLFxGraphAsync({
      lockfilePathUnderFixtures: '/website-sample-1/pnpm-lock-v6.0-rush.yaml',
      workspace: workspace
    });
    expect(serializedYaml).toMatchSnapshot();
  });
});
