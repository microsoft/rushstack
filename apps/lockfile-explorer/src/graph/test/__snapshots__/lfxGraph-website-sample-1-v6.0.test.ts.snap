// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lfxGraph-website-sample-1-v6.0 loads a workspace 1`] = `
"entries:
  - dependencies:
      - dependencyType: regular
        entryId: project:projects/d
        name: '@rushstack/d'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 3
        version: link:../d
    displayText: 'Project: a'
    entryId: project:projects/a
    entryPackageName: a
    entryPackageVersion: ''
    entrySuffix: ''
    jsonId: 0
    kind: 1
    packageJsonFolderPath: projects/a
    rawEntryId: ../../projects/a
    referrerJsonIds: []
    transitivePeerDependencies: []
  - dependencies:
      - dependencyType: regular
        entryId: project:projects/d
        name: '@rushstack/d'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 3
        version: link:../d
      - dependencyType: regular
        entryId: /@rushstack/n@2.0.0
        name: '@rushstack/n'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 9
        version: 2.0.0
    displayText: 'Project: b'
    entryId: project:projects/b
    entryPackageName: b
    entryPackageVersion: ''
    entrySuffix: ''
    jsonId: 1
    kind: 1
    packageJsonFolderPath: projects/b
    rawEntryId: ../../projects/b
    referrerJsonIds: []
    transitivePeerDependencies: []
  - dependencies:
      - dependencyType: regular
        entryId: project:projects/e
        name: '@rushstack/e'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 4
        version: link:../e
      - dependencyType: regular
        entryId: /@rushstack/k@1.0.0(@rushstack/m@1.0.0)(@rushstack/n@2.0.0)
        name: '@rushstack/k'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 6
        version: 1.0.0(@rushstack/m@1.0.0)(@rushstack/n@2.0.0)
      - dependencyType: regular
        entryId: /@rushstack/m@1.0.0
        name: '@rushstack/m'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 8
        version: 1.0.0
    displayText: 'Project: c'
    entryId: project:projects/c
    entryPackageName: c
    entryPackageVersion: ''
    entrySuffix: ''
    jsonId: 2
    kind: 1
    packageJsonFolderPath: projects/c
    rawEntryId: ../../projects/c
    referrerJsonIds: []
    transitivePeerDependencies: []
  - dependencies:
      - dependencyType: regular
        entryId: project:projects/e
        name: '@rushstack/e'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 4
        version: link:../e
      - dependencyType: regular
        entryId: /@rushstack/j@1.0.0(@rushstack/n@2.0.0)
        name: '@rushstack/j'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 5
        version: 1.0.0(@rushstack/n@2.0.0)
      - dependencyType: regular
        entryId: /@rushstack/n@2.0.0
        name: '@rushstack/n'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 9
        version: 2.0.0
    displayText: 'Project: d'
    entryId: project:projects/d
    entryPackageName: d
    entryPackageVersion: ''
    entrySuffix: ''
    jsonId: 3
    kind: 1
    packageJsonFolderPath: projects/d
    rawEntryId: ../../projects/d
    referrerJsonIds:
      - 0
      - 1
    transitivePeerDependencies: []
  - dependencies:
      - dependencyType: regular
        entryId: /@rushstack/n@3.0.0
        name: '@rushstack/n'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 10
        version: 3.0.0
    displayText: 'Project: e'
    entryId: project:projects/e
    entryPackageName: e
    entryPackageVersion: ''
    entrySuffix: ''
    jsonId: 4
    kind: 1
    packageJsonFolderPath: projects/e
    rawEntryId: ../../projects/e
    referrerJsonIds:
      - 2
      - 3
    transitivePeerDependencies: []
  - dependencies:
      - dependencyType: regular
        entryId: /@rushstack/k@1.0.0(@rushstack/m@1.0.0)(@rushstack/n@2.0.0)
        name: '@rushstack/k'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 6
        version: 1.0.0(@rushstack/m@1.0.0)(@rushstack/n@2.0.0)
      - dependencyType: regular
        entryId: /@rushstack/m@1.0.0
        name: '@rushstack/m'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 8
        version: 1.0.0
    displayText: '@rushstack/j 1.0.0 [@rushstack/n@2.0.0]'
    entryId: ''
    entryPackageName: '@rushstack/j'
    entryPackageVersion: 1.0.0
    entrySuffix: '@rushstack/n@2.0.0'
    jsonId: 5
    kind: 2
    packageJsonFolderPath: common/temp/node_modules/.pnpm/@rushstack+j@1.0.0_@rushstack+n@2.0.0/node_modules/@rushstack/j
    rawEntryId: /@rushstack/j@1.0.0(@rushstack/n@2.0.0)
    referrerJsonIds:
      - 3
    transitivePeerDependencies:
      - '@rushstack/n'
  - dependencies:
      - dependencyType: regular
        entryId: /@rushstack/l@1.0.0(@rushstack/m@1.0.0)(@rushstack/n@2.0.0)
        name: '@rushstack/l'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 7
        version: 1.0.0(@rushstack/m@1.0.0)(@rushstack/n@2.0.0)
    displayText: '@rushstack/k 1.0.0 [@rushstack/m@1.0.0; @rushstack/n@2.0.0]'
    entryId: ''
    entryPackageName: '@rushstack/k'
    entryPackageVersion: 1.0.0
    entrySuffix: '@rushstack/m@1.0.0; @rushstack/n@2.0.0'
    jsonId: 6
    kind: 2
    packageJsonFolderPath: >-
      common/temp/node_modules/.pnpm/@rushstack+k@1.0.0_@rushstack+m@1.0.0_@rushstack+n@2.0.0/node_modules/@rushstack/k
    rawEntryId: /@rushstack/k@1.0.0(@rushstack/m@1.0.0)(@rushstack/n@2.0.0)
    referrerJsonIds:
      - 2
      - 5
    transitivePeerDependencies:
      - '@rushstack/m'
      - '@rushstack/n'
  - dependencies:
      - dependencyType: regular
        entryId: /@rushstack/m@1.0.0
        name: '@rushstack/m'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 8
        version: 1.0.0
      - dependencyType: regular
        entryId: /@rushstack/n@2.0.0
        name: '@rushstack/n'
        peerDependencyMeta: {}
        resolvedEntryJsonId: 9
        version: 2.0.0
      - dependencyType: peer
        entryId: 'Peer: @rushstack/m'
        name: '@rushstack/m'
        peerDependencyMeta:
          name: '@rushstack/m'
          optional: false
          version: ^1.0.0
        version: ^1.0.0
      - dependencyType: peer
        entryId: 'Peer: @rushstack/n'
        name: '@rushstack/n'
        peerDependencyMeta:
          name: '@rushstack/n'
          optional: true
          version: ^2.0.0
        version: ^2.0.0
    displayText: '@rushstack/l 1.0.0 [@rushstack/m@1.0.0; @rushstack/n@2.0.0]'
    entryId: ''
    entryPackageName: '@rushstack/l'
    entryPackageVersion: 1.0.0
    entrySuffix: '@rushstack/m@1.0.0; @rushstack/n@2.0.0'
    jsonId: 7
    kind: 2
    packageJsonFolderPath: >-
      common/temp/node_modules/.pnpm/@rushstack+l@1.0.0_@rushstack+m@1.0.0_@rushstack+n@2.0.0/node_modules/@rushstack/l
    rawEntryId: /@rushstack/l@1.0.0(@rushstack/m@1.0.0)(@rushstack/n@2.0.0)
    referrerJsonIds:
      - 6
    transitivePeerDependencies: []
  - dependencies: []
    displayText: '@rushstack/m 1.0.0'
    entryId: ''
    entryPackageName: '@rushstack/m'
    entryPackageVersion: 1.0.0
    entrySuffix: ''
    jsonId: 8
    kind: 2
    packageJsonFolderPath: common/temp/node_modules/.pnpm/@rushstack+m@1.0.0/node_modules/@rushstack/m
    rawEntryId: /@rushstack/m@1.0.0
    referrerJsonIds:
      - 2
      - 5
      - 7
    transitivePeerDependencies: []
  - dependencies: []
    displayText: '@rushstack/n 2.0.0'
    entryId: ''
    entryPackageName: '@rushstack/n'
    entryPackageVersion: 2.0.0
    entrySuffix: ''
    jsonId: 9
    kind: 2
    packageJsonFolderPath: common/temp/node_modules/.pnpm/@rushstack+n@2.0.0/node_modules/@rushstack/n
    rawEntryId: /@rushstack/n@2.0.0
    referrerJsonIds:
      - 1
      - 3
      - 7
    transitivePeerDependencies: []
  - dependencies: []
    displayText: '@rushstack/n 3.0.0'
    entryId: ''
    entryPackageName: '@rushstack/n'
    entryPackageVersion: 3.0.0
    entrySuffix: ''
    jsonId: 10
    kind: 2
    packageJsonFolderPath: common/temp/node_modules/.pnpm/@rushstack+n@3.0.0/node_modules/@rushstack/n
    rawEntryId: /@rushstack/n@3.0.0
    referrerJsonIds:
      - 4
    transitivePeerDependencies: []
workspace:
  pnpmLockfileFolder: common/temp
  pnpmLockfilePath: common/temp/pnpm-lock.yaml
  rushConfig:
    rushVersion: 5.158.1
    subspaceName: ''
  workspaceRootFullPath: /repo
"
`;
